### PR TITLE
MULE-15908: Persistent ObjectStore doesn't regenerate the partition descriptor file after a clear

### PR DESCRIPTION
**Pull request by estebanwasinger**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/util/store/PersistentObjectStorePartition.java [hunk 1]
```diff
@@ -139,6 +139,7 @@ public void clear() throws ObjectStoreException {
     synchronized (realKeyToUUIDIndex) {
       try {
         cleanDirectory(this.partitionDirectory);
+        open();
```
Comment (alegmarra) [inline core/src/main/java/org/mule/runtime/core/internal/util/store/PersistentObjectStorePartition.java]: I think you only need the `createOrRetrievePartitionDescriptorFile();` part of the open

Comment (alegmarra) [reply core/src/main/java/org/mule/runtime/core/internal/util/store/PersistentObjectStorePartition.java]: also, does it make sense to delete the descriptor in the first place? or instead we need to whitelist the descriptor file?

File: core-tests/src/test/java/org/mule/runtime/core/internal/util/store/PersistentObjectStorePartitionTestCase.java [hunk 2]
```diff
@@ -100,11 +101,13 @@ public void clear() throws Exception {
...
 
     partition.store(KEY, VALUE);
-    assertTrue(partition.contains(KEY));
-    assertEquals(VALUE, partition.retrieve(KEY));
+    assertThat(partition.contains(KEY), is(true));
+    assertThat(VALUE, is(partition.retrieve(KEY)));
 
     partition.clear();
-    assertFalse(partition.contains(KEY));
+    assertThat(partition.contains(KEY), is(false));
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/internal/util/store/PersistentObjectStorePartitionTestCase.java]: This is testing the implementation not the behavior. What you should try to reproduce is that without your fix, writing to a store after clear fails and that afterwards you can successfully write and read from a cleared object store

File: core/src/main/java/org/mule/runtime/core/internal/util/store/PersistentObjectStorePartition.java (hunk 1 above)
Comment (marianogonzalez) [reply core/src/main/java/org/mule/runtime/core/internal/util/store/PersistentObjectStorePartition.java]: Check if mule 3 doesn't have the same bug. My guess is it does

**Pull request merged**